scenarios/kubernetes_e2e.py

```diff
@@ -572,7 +572,7 @@ def main(args):
         set_up_kops_aws(mode.workspace, args, mode, cluster, runner_args)
     elif args.deployment == 'kops' and args.provider == 'gce':
         set_up_kops_gce(mode.workspace, args, mode, cluster, runner_args)
-    elif args.gce_ssh:
+    elif args.gce_ssh and args.provider != 'local':
```

Comment:
this change doesn't seem right

Reply:
I'm up for suggestions on how best to address this. What's happening here is that args.gce_ssh seems to be defaulted if it's not set explicitly, so the existing "elif args.gce_ssh" is always true, and an attempt is made to add GCE SSH keys. One possible way to fix this would be to change the conditional to "elif args.provider == 'gce' or args.provider == 'gcp'", but I don't know if that's going to break some kubernetes-anywhere tests. Or if there's a way to NOT default args.gce_ssh without breaking other tests? In my case (and any other CI that uses kubeadm_e2e in a local or dind mode), I want it to skip setting up the SSH keys.

Reply:
cc @dims for this change